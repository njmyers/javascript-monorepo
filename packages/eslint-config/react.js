module.exports = {
  parser: 'babel-eslint',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  env: {
    'shared-node-browser': true,
  },
  extends: ['plugin:react/recommended'],
  plugins: ['react', 'react-hooks', 'jsx-a11y'],
  rules: {
    'jsx-a11y/label-has-associated-control': 'error',
    'react/display-name': 'off',
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
};
