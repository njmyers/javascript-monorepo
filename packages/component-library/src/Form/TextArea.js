// @flow
import * as React from 'react';
import FormElement from './FormElement';

const TextArea = React.forwardRef((props, ref) => (
  <FormElement as="textarea" forwardedRef={ref} {...props} />
));

TextArea.defaultProps = {
  name: 'text',
  element: 'textArea',
  required: false,
  block: 'formTextArea',
  placeholder: '',
};

export default TextArea;
