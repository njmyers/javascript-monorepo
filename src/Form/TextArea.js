// @flow
import * as React from 'react';
import BEM from '../BEM';

type Props = {
  name: string,
  label?: string,
  required: boolean,
  block: string,
  modifiers?: string | Array<string> | void,
  className: string,
  focus: boolean,
  onChange?: Function,
  placeholder?: string,
  value?: string | number,
  rows?: number,
  cols?: number,
};

class TextArea extends React.PureComponent<Props> {
  static defaultProps = {
    name: 'text',
    required: false,
    block: 'formTextArea',
    focus: false,
    placeholder: '',
  };

  // ref to this input
  input: ?HTMLTextAreaElement;
  // label defaults to property name
  label = this.props.label ? this.props.label : this.props.name;

  componentDidMount() {
    // $FlowFixMe
    if (this.props.focus) this.input.focus();
  }

  render() {
    return (
      <BEM block={this.props.block}>
        <div modifiers={this.props.modifiers}>
          <label
            element="label"
            modifiers={this.props.modifiers}
            htmlFor={this.props.name}
          >
            {this.label}
          </label>
          <textarea
            ref={(node) => (this.input = node)}
            element="textArea"
            modifiers={this.props.modifiers}
            value={this.props.value}
            name={this.props.name}
            onChange={this.props.onChange}
            required={this.props.required}
            placeholder={this.props.placeholder}
            rows={this.props.rows}
            cols={this.props.cols}
          />
        </div>
      </BEM>
    );
  }
}

export default TextArea;
