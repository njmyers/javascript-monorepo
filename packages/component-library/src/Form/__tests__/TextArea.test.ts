import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import TextArea from '../TextArea';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<TextArea />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
