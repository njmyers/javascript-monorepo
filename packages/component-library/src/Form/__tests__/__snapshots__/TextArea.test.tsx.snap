// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads without crashing 1`] = `
<div>
  <label
    element="label"
    htmlFor="text"
  >
    text
  </label>
  <textarea
    element="textArea"
    name="text"
    placeholder=""
    required={false}
  />
</div>
`;
